test/built-ins/Array/from/source-object-length.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2015 Microsoft Corporation. All rights reserved.
+// This code is governed by the license found in the LICENSE file.
+
+/*---
+description: >
+    Source is an object with length property and one item is deleted
+    from the source
+es6id: 22.1.2.1
+---*/
+
+var array = [2, 4, 0, 16];
+var expectedArray = [2, 4, , 16];
+var obj = {
+    length : 4,
+    0 : 2,
+    1 : 4,
+    2 : 0,
+    3 : 16
+}
+delete obj[2];
+var a = Array.from.call(Object, obj);
+assert.sameValue(typeof(a), "object", "The returned type is expected to be object.");

```

Comment:
Based on the invocation of `Array.from`, this appears to be a test for the use of a constructor other than `Array`. That's an important feature to test, but it should be done in a dedicated file with a name/frontmatter that describe the feature specifically.

Also, the expectation should be expressed in a way that would fail if the Array constructor were used incorrectly (note that `typeof Array.from([])` is *also* "object", so it would be better to check `Array.from.call(Object, []).constructor`)

By the way: since `typeof` is an operator, you do not need to wrap the operand in parenthesis.